server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/DatafileTransaction.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.tserver.tablet;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.accumulo.core.metadata.StoredTabletFile;
+
+public abstract class DatafileTransaction {
+
+  protected final long ts = System.currentTimeMillis();
+
+  public void apply(Set<StoredTabletFile> files) {}
```

Comment:
I'm not sure I understand the need to compute what the final set of files should be using this transaction log. I'm wondering if the transaction log is just TreeMap<Timestamp, DatafileTransaction>. When tablet memory does not match `datafileSizes`, dump the TreeMap to the log and clear it. Thoughts?

Reply:
The point of the pre-computation of the final set was to allow the checkTransactionLog in the DatafileManager to be as quick as possible.  This was for performance reasons.

Reply:
That makes sense if you are going to call `checkTransactionLog` after every modification of `datafileSizes`. If `checkTransactionLog` is called in the tablet health check thread, then the computation of the final set of files from the transaction log could be done at that time.

Reply:
I am calling checkTransactionLog after every modification of datafileSizes.....

Reply:
> per Keith's request.....

ah, I didn't realize that.